### PR TITLE
RD-3902 Use different 'getting started' modal configuration for initial screen and cloud setup button

### PR DESCRIPTION
**Pull request by kubama**
## Description
Introduced different 'getting started' modal configurations for initial screen and cloud setup button.
Still TODO in this PR:
* [x] replace https://www.zadara.com/wp-content/uploads/cloudlogos3.png link with a link to https://cloudify.co/wp-content/uploads once this image is uploaded there (waiting for HD to complete this task)
* [x] update related system test as it is failing now

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [X] I verified if that change requires a documentation update.
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/1937/

 

## Documentation
I assume no update is needed, please correct me if I'm wrong.

### CODE REVIEW

**Review by qooban — COMMENTED** [review]
As for now one comment from my side.
I'll try to quickly test it, but I cannot promise I'll manage.

File: app/components/GettingStartedModal/schema/gettingStarted.schema.json [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        "zipUrl": "https://repository.cloudifysource.org/cloudify/blueprints/6.2/terraform/terraform_openstack_vm.zip",
+        "yamlFile": "blueprint.yaml"
+      },
+      {
+        "id": "terraform_aws_vm",
+        "name": "Terraform-AWS-VM",
+        "zipUrl": "https://repository.cloudifysource.org/cloudify/blueprints/6.2/terraform/terraform_aws_vm.zip",
+        "yamlFile": "blueprint.yaml"
+      }
+    ]
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/components/GettingStartedModal/schema/gettingStarted.schema.json:40]: Probably.
@alexmolev Would you like to comment on that?

[the remark this answers is not in the document]
Comment (kubama) [reply app/components/GettingStartedModal/schema/gettingStarted.schema.json]: Looks like all blueprints used here are available for 6.3 - I updated the URLs

**Comment by qooban** [comment]
> I assume no update is needed, please correct me if I'm wrong.

I'm not aware of Getting Started documentation, so I think we are good here.

**Review by alexmolev — APPROVED** [review]

File: app/components/GettingStartedModal/useGettingStartedUrlParam.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import useSearchParam from '../../utils/hooks/useSearchParam';
+
+const gettingStartedParameterName = 'gettingStarted';
```
Comment (qooban) [inline app/components/GettingStartedModal/useGettingStartedUrlParam.ts:3]: Depends on time:

As I see right now the naming became a bit misleading. "gettingStarted" parameter is used to show cloud account setup and new set of environments is used as getting started.

File: app/components/GettingStartedModal/GettingStartedModal.tsx [hunk 3]
```diff
@@ -190,6 +198,7 @@ const GettingStartedModal = () => {
                 secretsStepsData={secretsStepsData}
                 secretsStepIndex={secretsStepIndex}
                 summaryStepSchemas={summaryStepSchemas}
+                schema={schema}
                 onEnvironmentsStepChange={handleEnvironmentsStepChange}
                 onSecretsStepChange={handleSecretsStepChange}
                 onInstallationStarted={handleInstallationStarted}
```
Comment (qooban) [inline app/components/GettingStartedModal/GettingStartedModal.tsx:204]: Depends on time:

This comment refers to L208:
```
{stepName !== StepName.Welcome && (
```

I think that "Don't show next time" checkbox is right now only relevant when we are showing Getting Started version (new "schema") and not Cloud Setup version (old "schema").

File: test/cypress/integration/getting_started_spec.ts [hunk 4]
```diff
@@ -142,7 +142,8 @@ describe('Getting started modal', () => {
 
     // NOTE: Most of the tests are wrapped inside another describe, so that we wouldn't have to resign from using the usePageMock. Because of that we can be granted performance boost, which is making test scenarios faster by approximately 5s.
     describe('', () => {
```
Comment (qooban) [inline test/cypress/integration/getting_started_spec.ts:144]: Depends on time:

Maybe worth to add a simple test to check whether proper data source is displayed? (either Cloud Setup or Getting Started)

**Review by qooban — COMMENTED** [review]
I did UI tests and it seems all working fine 👍🏻

I've added 3 comments. I think it is worth to address them, but all of them I annotated with "Depends on time" as I'm not sure if it is for this PR (=> 6.3 release) or for post-6.3 release.

**Review by NorbertAmartus — DISMISSED** [review]
Looks good to me 🎉

File: app/components/GettingStartedModal/useGettingStartedUrlParam.ts (hunk 2 above)
Comment (kubama) [reply app/components/GettingStartedModal/useGettingStartedUrlParam.ts]: Done

File: app/components/GettingStartedModal/GettingStartedModal.tsx (hunk 3 above)
Comment (kubama) [reply app/components/GettingStartedModal/GettingStartedModal.tsx:204]: Done

File: test/cypress/integration/getting_started_spec.ts (hunk 4 above)
Comment (kubama) [reply test/cypress/integration/getting_started_spec.ts:144]: Done

**Comment by kubama** [comment]
I started fresh system test execution - updated link in the description

**Review by NorbertAmartus — APPROVED** [review]

**Review by qooban — APPROVED** [review]
Looks great! Thanks for addressing all of my comments.

**Pull request merged**